### ISSUE TITLE
rightSection prop for PasswordInput have no effect

### ISSUE THREAD

**Issue by eriolo** [mantinedev/mantine#4436]
### What package has an issue

@mantine/core

### Describe the bug

Hi,

I noticed that the `rightSection` for the `PasswordInput` doesn't render, e.g. I cannot see the "Right section" in the DOM.

```
<PasswordInput
  label="Password"
  placeholder="Set a password"
  rightSection={<div>Right section</div>}
/>
```

### What version of @mantine/hooks page do you have in package.json?

^6.0.13

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/green-leftpad-yztjkq

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
PasswordInput rightSection is occupied by password visibility toggle control. If you need right section, use `<TextInput type="password" />` instead.

**Comment by mrbnclt** [comment]
The docs might have caused this confusion as the props `rightSection` is included in the `PasswordInput component props` [page](https://mantine.dev/core/password-input/?t=props#:~:text=on%20input%20element-,rightSection,-ReactNode). 

It might be better to remove the `rightSection` props in the docs as it is not really used by `PasswordInput` component.

**Comment by rtivital** [comment]
Fixed in 6.0.17
